vm/src/air_public_input.rs

```diff
@@ -0,0 +1,105 @@
+use std::collections::HashMap;
+
+use felt::Felt252;
+use serde::Serialize;
+use thiserror::Error;
+
+use crate::{
+    types::layout::CairoLayout,
+    vm::{
+        errors::{trace_errors::TraceError, vm_errors::VirtualMachineError},
+        trace::trace_entry::TraceEntry,
+    },
+};
+
+#[derive(Serialize, Debug)]
+pub struct PublicMemoryEntry {
+    address: usize,
+    page: usize,
+    value: Option<Felt252>,
```

Comment:
When serializing to json, the inner felt representation is being used:
```json
{
      "address": 1,
      "page": 0,
      "value": {
        "value": {
          "val": [
            2147450879,
            67600385
          ]
        }
      }
    },
```
vs cairo-lang implementation using hexadecimal string representation:
```json
{
            "page": 0,
            "address": 1,
            "value": "0x40780017fff7fff"
        },
```
We coud add a `#[serde(serialize_with = "path")]` attribute to the `value` field in order to serialize it as a hex string

Reply:
done!